### ISSUE TITLE
[BUG] CNNClassifier loss and accuracy flat 0.5 on all epochs

### ISSUE THREAD

**Issue by equ1** [sktime/sktime#3806]
**Describe the bug**
Running `CNNClassifier` with nested_univ data type (multivariate time series) gives a loss/accuracy of 0.5 on each epoch. Each iteration takes 4 ms, and I highly suspect no actual training is happening. Here's screencaps of what's going on:

![image](https://user-images.githubusercontent.com/19189325/202454225-7c334cc3-0a3f-45de-9b2b-9ecf36ee8420.png)
![image](https://user-images.githubusercontent.com/19189325/202454282-15dd4ad3-408d-428b-b0ab-4cfd8049025d.png)
![image](https://user-images.githubusercontent.com/19189325/202454340-86eb4b40-b369-4733-9c94-c5681ffcb580.png)

**To Reproduce**
Generate a nested_univ dataframe X_train with dimensions 400x8, with a 10000 length series within each cell (datetime index). Then generate another dataframe X_test of size 100x8 in the same fashion. y_train and y_test are numpy arrays of size (400,) and (100,) respectively. Then run the regular training process:

```python
from sktime.classification.deep_learning.cnn import CNNClassifier

network = CNNClassifier(n_epochs=250, verbose=True)
network.fit(X_train, y_train)
network.score(X_test, y_test)
```

**Expected behavior**
The model should train normally. I don't think there should be any GPU memory issues - I'm using an A4000 with 16 GiBs of VRAM and there are only 27452 trainable parameters.

**Versions**
sktime version = 0.14.0 (`pip install sktime[dl]`)
tensorflow version = 2.9.1
keras version = 2.9.0
tensorflow-probability version = 0.17.0 (manually downgraded from 0.18.0 because of compatibility issues)
keras-self-attention version  = 0.51.0 (installed separately because of dependency issue. Maybe this should be included in sktime[dl]?)

**Comment by TonyBagnall** [comment]
hi, thanks for highlighting this. In my experience, deep learners like CNN quite often fail completely and just predict a single class. All I can say is that for random data and for basic motions, it does something. For random data it gets 45% accuracy on the test (so is doing something!), but for BasicMotions it gets 100%. 

```
def _debug_cnn_not_fitting_3806(n_cases=20, n_dims=2, series_length=20):
    from sktime.classification.deep_learning import CNNClassifier

    trainX = np.random.rand(n_cases, n_dims, series_length)
    trainY = np.random.randint(0, 2, n_cases)
    testX = np.random.rand(n_cases, n_dims, series_length)
    testY = np.random.randint(0, 2, n_cases)
    print(trainY)
    cnn = CNNClassifier(n_epochs=250, verbose=True)
    print(" beginning fit to random")
    cnn.fit(trainX, trainY)
    d=cnn.score(testX,testY)
    print(" score =" , d)
    from datasets import load_basic_motions
    trainX, trainY = load_basic_motions(split="train")
    testX, testY = load_basic_motions(split="test")
    print(" beginning fit to basic motions")
    cnn.fit(trainX, trainY)
    d=cnn.score(testX,testY)
    print(" score =" , d)
```

**Comment by equ1** [comment]
That's strange...I manually implemented an RNN with Pytorch for the same dataset and it was reaching up to 90% training accuracy. I find it a little hard to believe that a 4ms iteration is actually doing some kind of processing considering it took my RNN hours to train. Maybe it's just the model architecture! Interesting stuff.

**Comment by equ1** [comment]
@TonyBagnall would like to add that documentation for all the deep learning functions is not included in the latest version (0.14.0). E.g. https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.classification.deep_learning.mlp.MLPClassifier.html.

**Comment by TonyBagnall** [comment]
thanks @equ1 I'll reopen if thats ok, what you describe does sound odd and worth investigating. Have you tried it inputting a numpy array, [n_cases][n_dimension][series_length] (I think [400][8][10000] for your example?

**Comment by TonyBagnall** [comment]
@achieveordie @AurumnPegasus any ideas/thoughts on this?

**Comment by achieveordie** [comment]
The first thing that I would suggest trying is to overfit the data - maybe consider a subset of your dataset and a comparatively larger estimator size. If the estimator overfits then probably the problem might be the current estimator doesn't have enough parameters to improve. If it doesn't.. then we'd have to think from a different perspective.

@equ1 just curious, what was the size of your RNN-based network?
